tests/test_read_only_trx.cpp

```diff
@@ -102,15 +102,23 @@ void test_trxs_common(std::vector<const char*>& specific_args, bool test_disable
    std::promise<std::tuple<producer_plugin*, chain_plugin*>> plugin_promise;
    std::future<std::tuple<producer_plugin*, chain_plugin*>> plugin_fut = plugin_promise.get_future();
    std::thread app_thread( [&]() {
-      fc::logger::get(DEFAULT_LOGGER).set_log_level(fc::log_level::debug);
-      std::vector<const char*> argv = {"test", "--data-dir", temp_dir_str.c_str(), "--config-dir", temp_dir_str.c_str()};
-      argv.insert( argv.end(), specific_args.begin(), specific_args.end() );
-      app->initialize<chain_plugin, producer_plugin>( argv.size(), (char**) &argv[0] );
-      app->find_plugin<chain_plugin>()->chain();
-      app->startup();
-      plugin_promise.set_value( {app->find_plugin<producer_plugin>(), app->find_plugin<chain_plugin>()} );
-      app->exec();
+      try {
+         fc::logger::get(DEFAULT_LOGGER).set_log_level(fc::log_level::debug);
+         std::vector<const char*> argv = {"test", "--data-dir", temp_dir_str.c_str(), "--config-dir", temp_dir_str.c_str()};
+         argv.insert(argv.end(), specific_args.begin(), specific_args.end());
+         app->initialize<chain_plugin, producer_plugin>(argv.size(), (char**)&argv[0]);
+         app->find_plugin<chain_plugin>()->chain();
+         app->startup();
+         plugin_promise.set_value({app->find_plugin<producer_plugin>(), app->find_plugin<chain_plugin>()});
+         app->exec();
+         return;
+      } FC_LOG_AND_DROP()
+      BOOST_CHECK(!"app threw exception see logged error");
    } );
+   fc::scoped_exit<std::function<void()>> on_except = [&](){
+      if (app_thread.joinable())
```

Comment:
Instread of having these two `app_thread.join();`, why not put line 102 to 176 in a block and rely on the `scoped_exit` to join the app_thread?

Reply:
Done.